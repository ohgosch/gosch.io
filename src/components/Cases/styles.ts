import styled, { css } from 'styled-components';
import { rem } from 'polished';

import { mobile, tabletDesktop } from 'visual/medias';
import Heading1 from 'visual/styles/Heading1';
import Heading2 from 'visual/styles/Heading2';
import Wrapper from 'visual/styles/Wrapper';
import Button from 'visual/styles/Button';

export const Container = styled.section`
  ${({ theme }) => css`
    background-color: ${theme.colors.white};
    padding: ${rem(64)} 0;
    color: ${theme.colors.black};
    overflow: hidden;

    ${Heading1} {
    }
  `}
`;

export const Content = styled(Wrapper)`
  display: grid;
  margin-top: ${rem(64)};

  ${tabletDesktop(css`
    grid-template-columns: repeat(2, 1fr);
    grid-gap: ${rem(80)};
  `)}

  ${mobile(css`
    grid-template-columns: 1fr 0.6fr;
    grid-gap: ${rem(64)};
  `)}
`;

type ColumnProps = {
  right?: boolean;
};

export const Column = styled.div<ColumnProps>`
  display: grid;
  position: relative;
  align-items: center;
  min-height: ${rem(422)};

  ${({ right }) => css`
    ${right &&
    css`
      text-align: right;

      ${Heading2} {
        margin-top: ${rem(40)};
      }

      ${Button} {
        margin-top: ${rem(40)};
      }
    `}
  `}
`;

export const ColumnContent = styled.div`
  position: absolute;
  width: ${rem(785)};

  img {
    border-radius: ${rem(8)};
  }

  //div div {
  //  max-width: auto !important;
  //}
`;

export const Features = styled.ul`
  list-style-type: none;
  margin-top: ${rem(16)};
`;

export const Feature = styled.li`
  font-style: normal;
  font-weight: normal;
  font-size: ${rem(20)};
  line-height: ${rem(28)};
  cursor: default;
`;
