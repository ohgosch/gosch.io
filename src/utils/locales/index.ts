import I18nextBrowserLanguageDetector from 'i18next-browser-languagedetector';
import I18NextLocizeBackend from 'i18next-locize-backend';
import { initReactI18next } from 'react-i18next';
import i18n from 'i18next';

import ptBR from './ptBR.json';
import en from './en.json';

i18n
  .use(I18NextLocizeBackend)
  .use(initReactI18next)
  .use(I18nextBrowserLanguageDetector)
  .init({
    debug: false,
    fallbackLng: 'en',
    whitelist: ['pt', 'en'],
    supportedLngs: ['pt', 'en'],
    detection: {
      order: ['querystring', 'navigator'],
      lookupQuerystring: 'lng',
    },
    resources: {
      en: {
        translation: en,
      },
      pt: {
        translation: ptBR,
      },
    },
  });

export default i18n;
