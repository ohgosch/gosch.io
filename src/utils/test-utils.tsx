import { ThemeProvider } from 'styled-components';
import { render } from '@testing-library/react';
import { ReactElement } from 'react';

import theme from 'visual/theme';

const customRender = (ui: ReactElement) =>
  render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>);

export * from '@testing-library/react';
export { customRender as render };
