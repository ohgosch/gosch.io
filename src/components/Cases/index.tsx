import Image from 'next/image';

import Heading1 from 'visual/styles/Heading1';
import Heading2 from 'visual/styles/Heading2';
import Button from 'visual/styles/Button';

import * as S from './styles';

const Cases = () => {
  return (
    <S.Container id="cases">
      <Heading1>Featured case</Heading1>
      <S.Content>
        <S.Column right>
          <div>
            <Image
              src="/assets/img/clients/bap.png"
              width={132}
              height={150}
              alt="Beer Awards Platform"
            />
          </div>
          <Heading2>Site Features</Heading2>
          <S.Features>
            <S.Feature>Discovery</S.Feature>
            <S.Feature>Site Prototype</S.Feature>
            <S.Feature>Development</S.Feature>
            <S.Feature>SEO Analytics</S.Feature>
          </S.Features>
          <div>
            <Button
              as="a"
              href="https://beerawardsplatform.com/home/"
              target="_blank"
            >
              Access published
            </Button>
          </div>
        </S.Column>
        <S.Column>
          <S.ColumnContent>
            <div>
              <Image
                src="/assets/img/clients/bap-screenshot.png"
                width={2072}
                height={1218}
              />
            </div>
          </S.ColumnContent>
        </S.Column>
      </S.Content>
    </S.Container>
  );
};

export default Cases;
