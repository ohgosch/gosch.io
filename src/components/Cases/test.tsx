import { render, screen } from 'utils/test-utils';

import Cases from '.';

describe('<Cases />', () => {
  it('should render correctly', () => {
    const { container } = render(<Cases />);

    expect(
      screen.getByRole('heading', { name: /Featured case/i }),
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
