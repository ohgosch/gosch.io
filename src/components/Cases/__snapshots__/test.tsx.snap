// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cases /> should render correctly 1`] = `
.c1 {
  font-weight: bold;
  text-align: center;
  -webkit-letter-spacing: 0.08em;
  -moz-letter-spacing: 0.08em;
  -ms-letter-spacing: 0.08em;
  letter-spacing: 0.08em;
  text-transform: uppercase;
  cursor: default;
}

.c6 {
  font-weight: bold;
  font-size: 2rem;
  line-height: 2.3125rem;
  -webkit-letter-spacing: 0.08em;
  -moz-letter-spacing: 0.08em;
  -ms-letter-spacing: 0.08em;
  letter-spacing: 0.08em;
  text-transform: uppercase;
  cursor: default;
}

.c10 {
  height: 2.5rem;
  padding: 0 1.5rem;
  font-style: normal;
  font-weight: bold;
  font-size: 0.875rem;
  color: #F2F2F2;
  background-color: #DBA14B;
  border-radius: 0.25rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2 {
  margin-left: auto;
  margin-right: auto;
}

.c0 {
  background-color: #F2F2F2;
  padding: 4rem 0;
  color: #171717;
  overflow: hidden;
}

.c3 {
  display: grid;
  margin-top: 4rem;
}

.c4 {
  display: grid;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 26.375rem;
  text-align: right;
}

.c4 .c5 {
  margin-top: 2.5rem;
}

.c4 .c9 {
  margin-top: 2.5rem;
}

.c11 {
  display: grid;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 26.375rem;
}

.c12 {
  position: absolute;
  width: 49.0625rem;
}

.c12 img {
  border-radius: 0.5rem;
}

.c7 {
  list-style-type: none;
  margin-top: 1rem;
}

.c8 {
  font-style: normal;
  font-weight: normal;
  font-size: 1.25rem;
  line-height: 1.75rem;
  cursor: default;
}

@media (min-width:46.0625rem) {
  .c1 {
    font-size: 3rem;
    line-height: 3.5rem;
  }
}

@media (max-width:46rem) {
  .c1 {
    font-size: 2.75rem;
    line-height: 3.25rem;
  }
}

@media (max-width:46rem) {
  .c2 {
    width: 90%;
    max-width: 46rem;
  }
}

@media (min-width:46.0625rem) and (max-width:64rem) {
  .c2 {
    width: 88%;
    max-width: 64rem;
  }
}

@media (min-width:64.0625rem) {
  .c2 {
    width: 90%;
    max-width: 64.0625rem;
  }
}

@media (min-width:46.0625rem) {
  .c3 {
    grid-template-columns: repeat(2,1fr);
    grid-gap: 5rem;
  }
}

@media (max-width:46rem) {
  .c3 {
    grid-template-columns: 1fr 0.6fr;
    grid-gap: 4rem;
  }
}

<section
  class="c0"
  id="cases"
>
  <h2
    class="c1"
  >
    Featured case
  </h2>
  <div
    class="c2 c3"
  >
    <div
      class="c4"
    >
      <div>
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTMyIiBoZWlnaHQ9IjE1MCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
              style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
            />
          </div>
          <noscript />
          <img
            alt="Beer Awards Platform"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </div>
      </div>
      <h3
        class="c5 c6"
      >
        Services
      </h3>
      <ul
        class="c7"
      >
        <li
          class="c8"
        >
          Discovery
        </li>
        <li
          class="c8"
        >
          Site Prototype
        </li>
        <li
          class="c8"
        >
          Development
        </li>
        <li
          class="c8"
        >
          SEO Analytics
        </li>
      </ul>
      <div>
        <a
          class="c9 c10"
          href="https://beerawardsplatform.com/home/"
          target="_blank"
        >
          Access published
        </a>
      </div>
    </div>
    <div
      class="c11"
    >
      <div
        class="c12"
      >
        <div>
          <div
            style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="box-sizing: border-box; display: block; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjA3MiIgaGVpZ2h0PSIxMjE4IiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIvPg=="
                style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
              />
            </div>
            <noscript />
            <img
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
